### PR TITLE
Oo/cloudmagic/10191/prime cli can convert fhir to fhir

### PR DESCRIPTION
**Pull request by cloudmagic80**
This PR is for the  Prime CLI can convert FHIR to FHIR using sender transforms

Test Steps:

```
./prime fhirdata --input-file src/testIntegration/resources/datatests/FHIR_to_FHIR/sample_AOE_1_20230220-0001.fhir -s ./metadata/fhir_transforms/senders/default-sender-transform.yml --output-format FHIR --output-file src/testIntegration/resources/datatests/FHIR_to_FHIR/sample_AOE_1_20230220-0001-expected.fhir
```

## Changes

1. Added convertToFhirToFhir function inside `ProcessFhirCommands.kt` and renamed fhirToHl7Schema variable to transform schema since `-s` option can take in any schema
2. Added FHIR_to_FHIR folder under  datasets and ran the test command above

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #10191 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by victor-chaparro** [comment]
@cloudmagic80 Can you fill in the rest of the PR template? Especially the Test Steps section, so that anybody that picks up this PR knows how to test it.

**Comment by oslynn** [comment]
Please run the following tests.
image

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 1]
```diff
@@ -106,26 +113,41 @@ class ProcessFhirCommands : CliktCommand(
...
             else -> {
                 val bundle = FhirTranscoder.decode(jsonString)
                 FhirToHl7Converter(
-                    fhirToHl7Schema!!.name.split(".")[0], fhirToHl7Schema!!.parent,
-                    context = FhirToHl7Context(
-                        CustomFhirPathFunctions(),
-                        null,
-                        translationFunctions = CustomTranslationFunctions()
-                    )
+                    transformSchema!!.name.split(".")[0], transformSchema!!.parent,
```
Comment (victor-chaparro) [inline prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:125]: Why did we change this? This should stay.

Comment (kant777) [reply prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:125]: was a merging issue. Fixed now

**Pull request merged**